lms/djangoapps/certificates/queue.py

```diff
@@ -306,6 +307,15 @@ def add_cert(self, student, course_id, course=None, forced_grade=None, template_
                 #   Despite blowing up the xml parser, bad values here are fine
                 grade_contents = None
 
+            # If this user's enrollment is not eligible to receive a
+            # certificate, mark it as such for reporting and
+            # analytics.
+            if not is_whitelisted and not CourseMode.is_eligible_for_certificate(enrollment_mode):
+                cert.eligible_for_certificate = False
+                new_status = CertificateStatuses.auditing
```

Comment:
also, not in the diff, but if we clean this up can we also clean up the same pattern in the method below.